src/controllers/EventController.ts

```diff
@@ -75,4 +81,35 @@ export class EventController {
 
     return this.eventMapper.entityToResponse(deletedEvent);
   }
+
+  @Post('/:eventID/signin')
+  @OnUndefined(409)
+  @ResponseSchema(Attendance)
+  async signInToEvent(
+    @Param('eventID') eventID: number,
+    @Body() appUserRequest: EventSignInRequest
+  ): Promise<Attendance | undefined> {
+    const { email } = appUserRequest;
+    const appUserFromEmail = await this.appUserService.getAppUserByEmail(email);
+    const appUserToSave = await this.appUserMapper.requestToEntityByEmail(
```

Comment:
hm a little weird to have mapper take in an entity if the mapping layer translates requests to entities can't we just pass in the request to the mapper and have it do the getAppUserByEmail stuff?

Reply:
Is it okay for mappers to use services then? If that's the case I'll change

Reply:
I'd say mappers can use services yeah - mappers and controllers sit at the same layer of req/res parsing stuff. if controllers can use services then mappers should be able to do so as well.

Reply:
Sounds good

Reply:
you bring up an interesting point though - if mappers are at the same layer as controllers then should they directly rely on repositories? (this is directed more at eventmapper)

lmk what you think and maybe we'll go ahead and change eventmapper to rely on services instead

Reply:
I think if we don't rely directly on repositories for mappers, then we'll have to add more CRUD operations to services. If we do, however, I feel like it might be a violation layer because services already use repositories directly, and if mappers are at the same level as controllers then mappers are at a higher layer than services.